R/dots-ellipsis.R

```diff
@@ -90,24 +90,25 @@ check_dots_unnamed <- function(env = caller_env(),
                                error = NULL,
                                call = caller_env(),
                                action = abort) {
-  proms <- ellipsis_dots(env, auto_name = FALSE)
+  proms <- ellipsis_dots(env)
   if (length(proms) == 0) {
     return()
   }
 
-  unnamed <- is.na(names(proms))
+  unnamed <- names2(proms) == ""
   if (all(unnamed)) {
     return(invisible())
   }
+  named <- !unnamed
 
-  named <- names(proms)[!unnamed]
   action_dots(
     error = error,
     action = action,
-    message = paste0(length(named), " arguments in `...` had unexpected names."),
-    dot_names = named,
+    message = "Arguments in `...` can't be named.",
```

Comment:
Must not be?

Reply:
Maybe it would be better to remove the negative and match the error above: `All arguments in `...` must be unnamed` ?